doc: release-notes-3.7: Add ADC release notes (#75054)
zephyrproject-rtos/zephyr: doc/releases/release-notes-3.7.rst

* ADC

>   * Added ADC_DT_SPEC_*BY_NAME() macros to get ADC io-channel information from DT by name.

Reviewer: ```suggestion
  * Added ``ADC_DT_SPEC_*BY_NAME()`` macros to get ADC IO-channel information from DT by name.
```
maybe better?

  * Added support for voltage biasing:
    * Added a :kconfig:option:`CONFIG_ADC_CONFIGURABLE_VBIAS_PIN` selected by drivers that support
      voltage biasing.

>     * Added a `zephyr,vbias-pins` to the adc-controller base binding to describe voltage

Reviewer 2: ```suggestion
    * Added a ``zephyr,vbias-pins`` property to the adc-controller base binding to describe voltage
```

      bias pins.

>     * Implemented for the ADC114s08 ADC driver.

Reviewer: ```suggestion
    * Implemented TI ADC114s08 ADC driver.
```
better?

Author: I can add TI but this is supposed to be a sub bullet of the voltage biasing addition, I am not sure if I'm doing it right

  * Renamed existing ADC sample to adc_dt.

>   * Added a new sample called adc_sequence that shows more of the runtime adc sequence features.

Reviewer: ```suggestion
  * Added a new sample called adc_sequence that shows more of the runtime ADC sequence features.
```

  * Added driver for the ENE KB1200.
  * Added driver for the NXP GAU ADC.

>   * Added I2C bus support for ad559x.

Reviewer: ```suggestion
  * Added I2C bus support for Analog Devices' AD559x.
```

>   * Added support for ad5593.

Reviewer: ```suggestion
  * Added support for Analog Devices' AD5593.
```

  * Added configuration of internal reference voltage value to ad559x to support

>     calls of adc_raw_to_millivolts().

Reviewer: ```suggestion
    calls of :c:func:`adc_raw_to_millivolts`.
```
better?

  * Fixed issue with driver initialization causing improper operation in the ad559x driver
    regarding the availibility of :kconfig:option:`CONFIG_THREAD_NAME`.

>   * Improved the adc read efficiency and validation in ad559x driver.

Reviewer: ```suggestion
  * Improved the ADC read efficiency and validation in ad559x driver.
```

>   * Updated esp32 ADC driver to work with version 5.1 of hal_espressif.

Reviewer: ```suggestion
  * Updated ESP32 ADC driver to work with version 5.1 of hal_espressif.
```

  * Added support for DMA mode operation for ESP32S3 and ESP32C3.
  * Fixed issue with configuration register access in the ads1119 driver.
  * Enabled acquisition time feature in the NXP LPADC driver.
  * Supported regulator output as reference to NXP LPADC.

>   * Fixed various issues with dma support in the STM32 ADC driver.

Reviewer: ```suggestion
  * Fixed various issues with DMA support in the STM32 ADC driver.
```

  * Fixed uninitialized value in kb1200 driver found in static analysis.

>   * Fixed issue with adc_raw_to_millivolts returning half the actual voltage with

Reviewer: ```suggestion
  * Fixed issue with :c:func:`adc_raw_to_millivolts` returning half the actual voltage with
```

    the tla2021 driver by correcting the reference voltage value.
  * Supported Nuvoton m2l31x in the numaker ADC driver.
  * Supported nRF54L15 and nRF54H20 in the nrfx_saadc driver.

>   * Improved the nrf saadc driver by disabling burst mode on unused channels, avoiding freezes.

Reviewer: ```suggestion
  * Improved the nRF SAADC driver by disabling burst mode on unused channels, avoiding freezes.
```

>   * Added support for power management to the smartbond sdadc and gpadc drivers.

Reviewer: ```suggestion
  * Added support for power management to the Smartbond SDADC and GPADC drivers.
```

>   * Fixed support for :kconfig:option:`CONFIG_PM_DEVICE_RUNTIME` in the smartbond ADC driver.

Reviewer: ```suggestion
  * Fixed support for :kconfig:option:`CONFIG_PM_DEVICE_RUNTIME` in the Smartbond ADC driver.
```

> 

Reviewer: ```suggestion
```

* ADC

  * Added ``ADC_DT_SPEC_*BY_NAME()`` macros to get ADC IO-channel information from DT by name.
  * Added support for voltage biasing:

>      * Added a :kconfig:option:`CONFIG_ADC_CONFIGURABLE_VBIAS_PIN` selected by drivers that support

Reviewer: An empty line above this one is needed

Author: @aescolar why?

Author: oh, I see there is a weird highlight

* ADC

  * Added ``ADC_DT_SPEC_*BY_NAME()`` macros to get ADC IO-channel information from DT by name.
  * Added support for voltage biasing:

    * Added a :kconfig:option:`CONFIG_ADC_CONFIGURABLE_VBIAS_PIN` selected by drivers that support
      voltage biasing.
    * Added a ``zephyr,vbias-pins`` property to the adc-controller base binding to describe voltage
      bias pins.
    * Implemented for the TI ADC114s08 ADC driver.
  * Renamed existing ADC sample to adc_dt.
  * Added a new sample called adc_sequence that shows more of the runtime

>     :c:struct:adc_sequence features.

Reviewer 3: ```suggestion
    :c:struct:`adc_sequence` features.
```

  * Added driver for the ENE KB1200.
  * Added driver for the NXP GAU ADC.
  * Added support for STM32H7R/S series.
  * Added support for ADI's ad5593.
  * Added I2C bus support for ADI ad559x.
  * Added configuration of internal reference voltage value to ad559x to support
    calls of :c:func:`adc_raw_to_millivolts()`.
  * Fixed issue with driver initialization causing improper operation in the ad559x driver
    regarding the availibility of :kconfig:option:`CONFIG_THREAD_NAME`.
  * Improved the ADC read efficiency and validation in ad559x driver.
  * Updated ESP32 ADC driver to work with version 5.1 of hal_espressif.
  * Added support for DMA mode operation for ESP32S3 and ESP32C3.
  * Fixed issue with configuration register access in the ads1119 driver.
  * Enabled acquisition time feature in the NXP LPADC driver.

>   * Supported regulator output as reference to NXP LPADC.

Reviewer 3: ```suggestion
  * Added support for regulator output as reference to NXP LPADC.
```

  * Changed phandle type DT property ``nxp,reference-supply`` to phandle-array type DT property
    ``nxp,references`` in ``nxp,lpc-lpadc`` binding. The NXP LPADC driver now supports passing
    the reference voltage value by using ``nxp,references``.
  * Fixed various issues with DMA support in the STM32 ADC driver.
  * Fixed uninitialized value in kb1200 driver found in static analysis.
  * Fixed issue with :c:func:`adc_raw_to_millivolts` returning half the actual voltage with
    the tla2021 driver by correcting the reference voltage value.

>   * Supported Nuvoton m2l31x in the numaker ADC driver.
>   * Supported nRF54L15 and nRF54H20 in the nrfx_saadc driver.

Reviewer 3: ```suggestion
  * Added support for Nuvoton m2l31x in the numaker ADC driver.
  * Added support for nRF54L15 and nRF54H20 in the nrfx_saadc driver.
```
